### ISSUE TITLE
Support Kafka Consumer in exact-once semantics

### ISSUE THREAD

**Issue by fx19880617** [apache/incubator-pinot#6480]
Seeing more requirements on this.

Kafka consumer supports exact-once semantics but requires code changes.
Code snippet:
```
KafkaConsumer consumer = createKafkaConsumer(
  "bootstrap.servers", "localhost:9092",
  "group.id", "my-group-id",
  "isolation.level", "read_committed");
consumer.subscribe(singleton("inputTopic"));
while (true) {
  ConsumerRecords records = consumer.poll(Long.MAX_VALUE);
  producer.beginTransaction();
  for (ConsumerRecord record : records)
    producer.send(producerRecord("outputTopic", record));
  producer.sendOffsetsToTransaction(currentOffsets(consumer), group);  
  producer.commitTransaction();
}
```

We can add a config in kafka-2.0 plugin and switch the code path.

**Comment by Jackie-Jiang** [comment]
@yupeng9 Thanks for adding the support. Can you please also add it in the documentation?

**Comment by yupeng9** [comment]
@Jackie-Jiang plz review it at https://github.com/pinot-contrib/pinot-docs/pull/24

**Comment by fx19880617** [comment]
Supported in PR #6580. This feature will be available in Pinot 0.7.0 release.
